src/Symfony/Component/Form/Extension/HttpFoundation/HttpFoundationRequestHandler.php

```diff
@@ -57,13 +60,13 @@ public function handleRequest(FormInterface $form, mixed $request = null)
             if ('' === $name) {
                 $data = $request->query->all();
             } else {
-                // Don't submit GET requests if the form's name does not exist
-                // in the request
-                if (!$request->query->has($name)) {
+                $missingData = $this->missingDataHandler->missingData;
+
+                if ($missingData === $data = $this->missingDataHandler->handle($form, $request->query->all()[$name] ?? $missingData)) {
```

Comment:
I cannot follow this code and would love if we could break this down into smaller chunks, maybe using expressive variable names.

Reply:
Done. Is it easier to follow?